__tests__/run.test.ts

```diff
@@ -1,7 +1,199 @@
+        jest.spyOn(os, 'type').mockReturnValue('Darwin');
+    
+        expect(run.getExecutableExtension()).toBe('');         
+        expect(os.type).toBeCalled();         
+    });
+
+    test('getKubectlPath() - return path to existing kubectl', async () => {
+        jest.spyOn(io, 'which').mockResolvedValue('pathToKubectl');
+
+        expect(await run.getKubectlPath()).toBe('pathToKubectl');
+        expect(io.which).toBeCalled();         
+    });
+
+    test('getKubectlPath() - throw error when kubectl not installed', async () => {
+        jest.spyOn(io, 'which').mockResolvedValue('');
+        jest.spyOn(toolCache, 'findAllVersions').mockReturnValue([]);
+
+        await expect(run.getKubectlPath()).rejects.toThrow('Kubectl is not installed');
+        expect(io.which).toBeCalled();         
+        expect(toolCache.findAllVersions).toBeCalled();         
+    });
+
+    test('getKubectlPath() - return path to kubectl in toolCache', async () => {
+        jest.spyOn(io, 'which').mockResolvedValue('');
+        jest.spyOn(toolCache, 'findAllVersions').mockReturnValue(['v1.15.0']);
+        jest.spyOn(toolCache, 'find').mockReturnValue('pathToTool');
+        jest.spyOn(os, 'type').mockReturnValue('Windows_NT');
+
+        expect(await run.getKubectlPath()).toBe(path.join('pathToTool', 'kubectl.exe'));
+        expect(io.which).toBeCalled();         
+        expect(toolCache.findAllVersions).toBeCalled();         
+        expect(toolCache.find).toBeCalled();         
+    });
+    
+    test('getKubeconfig() - throw error on invalid input', () => {
+        jest.spyOn(core, 'getInput').mockReturnValue('invalid');
+
+        expect(() => run.getKubeconfig()).toThrow('Invalid method specified. Acceptable values are kubeconfig and service-account.');
+        expect(core.getInput).toBeCalled();         
+    });
+
+    test('getKubeconfig() - return kubeconfig from input', () => {
+        jest.spyOn(core, 'getInput').mockImplementation((inputName, options) => {
+            if (inputName == 'method') return 'kubeconfig';
+            if (inputName == 'kubeconfig') return '###';
+        });
+        jest.spyOn(core, 'debug').mockImplementation();
+
+        expect(run.getKubeconfig()).toBe('###');
+        expect(core.getInput).toBeCalledWith('method', { required: true });         
+        expect(core.getInput).toBeCalledWith('kubeconfig', { required: true });         
+    });
+
+    test('getKubeconfig() - create kubeconfig from secret provided and return it', () => {
+        jest.spyOn(core, 'debug').mockImplementation();
+        const k8Secret = fs.readFileSync('__tests__/sample-secret.yml').toString();
+        const kubeconfig = JSON.stringify({
+            "apiVersion": "v1",
+            "kind": "Config",
+            "clusters": [
+                {
+                    "cluster": {
+                        "certificate-authority-data": 'LS0tLS1CRUdJTiBDRWyUSUZJQ',
```

Comment:
should this be part of secret?

Reply:
if you mean a github secret, no, just a dummy secret to represent what the k8s-secret object would look like if produced by the steps given in readme.